Fleshed out instructions about DD4hep geometry information (#164)
key4hep/k4MarlinWrapper: doc/starterkit/k4MarlinWrapperCLIC/Readme.md

The ``MarlinDD4hep::InitializeDD4hep`` processor can be replaced by the ``k4SimGeant4::GeoSvc`` and the
``TrackingCellIDEncodingSvc`` the latter of which is part of the k4MarlinWrapper repository.

> This requires removing the `InitDD4hep` object from the `algList` and adding `geoservice` and `cellIDSvc` to the `svcList`.

Reviewer: ```suggestion
This requires removing the wrapped `InitDD4hep` processor from the `algList` and adding `geoservice` and `cellIDSvc` to the `ExtSvc` in the `ApplicationMgr`.
```
I would use this slightly changed wording here, but maybe this is too specific for newcomers?

We can create the `geoservice` and `cellIDSvc` with;

```python
import os
from Gaudi.Configuration import INFO
from Configurables import GeoSvc, TrackingCellIDEncodingSvc

> geoservice = GeoSvc("GeoSvc")

Reviewer: ```suggestion

geoservice = GeoSvc("GeoSvc")
```

geoservice.detectors = [os.environ["K4GEO"]+"/CLIC/compact/CLIC_o3_v15/CLIC_o3_v15.xml"]
geoservice.OutputLevel = INFO
geoservice.EnableGeant4Geo = False

cellIDSvc = TrackingCellIDEncodingSvc("CellIDSvc")
cellIDSvc.EncodingStringParameterName = "GlobalTrackerReadoutID"
cellIDSvc.GeoSvcName = geoservice.name()
cellIDSvc.OutputLevel = INFO
```

> Then we need to make a `svcList` to add them to.

Reviewer: ```suggestion
Then we need to make a `svcList` to add them to and pass that to the `ExtSvc` argument.
```

At the bottom of the file, change 

```python
ApplicationMgr( TopAlg = algList,
                EvtSel = 'NONE',
                EvtMax   = 3,
                ExtSvc = [evtsvc],
                OutputLevel=WARNING
              )
```

to

```python

> svcList = [evtsvc]

Reviewer: ```suggestion
svcList = [evtsvc]  # if an evtsvc is available
```

The ``MarlinDD4hep::InitializeDD4hep`` processor can be replaced by the ``k4SimGeant4::GeoSvc`` and the
``TrackingCellIDEncodingSvc`` the latter of which is part of the k4MarlinWrapper repository.
This requires removing the wrapped `InitDD4hep` processor from the `algList` and adding `geoservice` and `cellIDSvc` to the `ExtSvc` in the `ApplicationMgr`.

For example:
We can create the `geoservice` and `cellIDSvc` with;

```python
import os
from Gaudi.Configuration import INFO
from Configurables import GeoSvc, TrackingCellIDEncodingSvc
svcList = []

geoservice = GeoSvc("GeoSvc")
geoservice.detectors = [os.environ["K4GEO"]+"/CLIC/compact/CLIC_o3_v15/CLIC_o3_v15.xml"]
geoservice.OutputLevel = INFO
geoservice.EnableGeant4Geo = False

> svcList.append(geoservice)

Reviewer 2: It is more modular to append the service to the svcList here, because if I want to get rid of the service, I just have to comment the block here, and not find it again below, otherwise one might as well just put `ExtSvc = [geoservice, cellIDSvc, evtsvc]` in the ApplicationMgr.

Author: Ok, in that case would it need to happen in the same location as `evtsvc = k4DataSvc('EventDataSvc')` is being created, since that needs to go in the list first? (I assume they don't commute)

Reviewer 2: That depends on what the "it" is that needs to happen?

But like this is:

https://github.com/key4hep/CLDConfig/blob/4cb3e121927352a36319910d95ef4cdcf5f8c483/CLDConfig/CLDReconstruction.py#L26-L30

Reviewer: I think the `evtsvc` could in principle go anywhere since it should be independent of the other services. However, I would just put it first since it is (almost) always present and we should try to keep things consistent in our examples and documentation.

Author: I've rearranged things, is that what you had in mind?

Reviewer 2: Yes, fine for me!
Thanks!

cellIDSvc.GeoSvcName = geoservice.name()
cellIDSvc.OutputLevel = INFO
```

Then we need to make a `svcList` to add them to and pass that to the `ExtSvc` argument.

> At the bottom of the file, change 

Reviewer: ```suggestion
At the bottom of the file, change
```

pre-commit complains about this space here.

In the space following

```python
from Configurables import k4DataSvc, PodioInput
evtsvc = k4DataSvc('EventDataSvc')
evtsvc.input = os.path.join('$TEST_DIR/inputFiles/', os.environ.get("INPUTFILE", "ttbar_edm4hep.root"))
```

we can start the `svcList` list and add the `GeoSvc` and `TrackingCellIDEncodingSvc` with;

```python
svcList = []
svcList.append(evtsvc)

import os
from Gaudi.Configuration import INFO
from Configurables import GeoSvc, TrackingCellIDEncodingSvc

geoservice = GeoSvc("GeoSvc")
geoservice.detectors = [os.environ["K4GEO"]+"/CLIC/compact/CLIC_o3_v15/CLIC_o3_v15.xml"]
geoservice.OutputLevel = INFO
geoservice.EnableGeant4Geo = False
svcList.append(geoservice)

cellIDSvc = TrackingCellIDEncodingSvc("CellIDSvc")
cellIDSvc.EncodingStringParameterName = "GlobalTrackerReadoutID"
cellIDSvc.GeoSvcName = geoservice.name()
cellIDSvc.OutputLevel = INFO
svcList.append(cellIDSvc)
```

Then all that is left is to pass that to the `ExtSvc` argument.
At the bottom of the file;

```python
ApplicationMgr( TopAlg = algList,
                EvtSel = 'NONE',
                EvtMax   = 3,
                ExtSvc = [evtsvc],
                OutputLevel=WARNING
              )
```

> changes to 

Reviewer: ```suggestion
changes to
```
